pkgs/misc/vim-plugins/overrides.nix

```
@@ -134,6 +135,12 @@ self: super: {
     };
   };
 
+  coc-prettier = buildVimPluginFrom2Nix {
+    pname = "coc-prettier";
+    version = nodePackages.coc-prettier.version;
+    src = nodePackages.coc-prettier + "/lib/node_modules/coc-prettier";
```

not sure if this is right, but I think you were trying to create a path.
```suggestion
    src = "${nodePackages.coc-prettier}/lib/node_modules/coc-prettier";
```